app/services/open_library_service.rb

```diff
@@ -0,0 +1,43 @@
+class OpenLibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org'
+
+  def initialize(isbn)
+    @isbn = isbn
+    @uri = '/api/books'
+    @options = {
+      'bibkeys': "ISBN:#{@isbn}",
+      'format': 'json',
+      'jscmd': 'data'
+    }
+  end
+
+  def book_info
+    raise 'Error fetching data from OpenLibrary API' if retrieve_data.code.to_i != 200
```

Comment:
It's not a good practice to compare directly to 200 status, for example it maybe 204 which is `no_content` and it's still a valid response, a good way to compare it that is a 2xx HTTP code will be to see if its lower than 200 and higher than 299